### PR TITLE
PT-3196: Deprecate SolrService and use the Vocabularies RESTful API instead

### PR DESCRIPTION
**Pull request by veronikaslc**

### CODE REVIEW

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml [hunk 1]
```diff
@@ -1489,7 +1489,7 @@ document.observe('xwiki:dom:loaded', function()
     initialize : function(dataTable) {
       /** Measurement evaluation service */
       this.measurementsService = new XWiki.Document('PatientToolsService', 'PhenoTips').getURL('get');
-      this.solarService = new XWiki.Document('SolrService', 'PhenoTips').getURL("get");
+      this.solrService = XWiki.contextPath + "/rest/vocabularies/hpo/suggest";
```
Comment (marta-) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml]: why did you rename it?

Comment (sdumitriu) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml]: It was badly named before.

Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml]: `solAr` just hurts to see

File: components/vocabularies/rest/src/main/java/org/phenotips/vocabularies/rest/internal/DefaultVocabularyTermSuggestionsResource.java [hunk 2]
```diff
@@ -79,18 +75,16 @@ public VocabularyTerms suggest(String vocabularyId, String input, @DefaultValue(
...
         for (VocabularyTerm term : termSuggestions) {
-            VocabularyTermSummary termRep = this.objectFactory.createVocabularyTermRepresentation(term);
-            termRep.withLinks(this.autolinker.get().forSecondaryResource(VocabularyTermResource.class, this.uriInfo)
-                .withActionableResources(VocabularyResource.class)
-                .withExtraParameters("vocabulary-id", vocabularyId)
-                .withExtraParameters("term-id", term.getId())
-                .build());
-            termReps.add(termRep);
+            JSONObject trm = term.toJSON();
+            trm.put("links", this.autolinker.get().forSecondaryResource(VocabularyTermResource.class, this.uriInfo));
```
Comment (a-gvo) [inline components/vocabularies/rest/src/main/java/org/phenotips/vocabularies/rest/internal/DefaultVocabularyTermSuggestionsResource.java]: I think you forgot ```.build()```?

Comment (veronikaslc) [inline components/vocabularies/rest/src/main/java/org/phenotips/vocabularies/rest/internal/DefaultVocabularyTermSuggestionsResource.java]: right

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml (hunk 1 above)
Comment (allasm) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml]: actually, since it is technically does not have to be solr-based any more wouldn't it be better to rename it to `hpoSuggestService` (or smth similar), which it is? I mean, from the client code point of view it does not mater what technology we use at the back-end

File: components/pedigree/resources/src/main/resources/pedigree/hpoTerm.js [hunk 3]
```diff
@@ -75,7 +75,7 @@ define([
     }
 
     HPOTerm.getServiceURL = function() {
```
Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/hpoTerm.js]: I would remove this and added `getHPOSuggestURL()` (or smth similar) to `externalEndpoints.js`

File: components/widgets/resources/src/main/resources/resources/uicomponents/PhenotypeSelectionUtils.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ define("PhenotypeSelectionUtils", [], function() {
            return null;
         }
         var id = phenotype.id;
-        var searchUrl = new XWiki.Document('SolrService', 'PhenoTips').getURL("get", "q=" + id);
+        var searchUrl = XWiki.contextPath + "/rest/vocabularies/hpo/" + id;
```
Comment (allasm) [inline components/widgets/resources/src/main/resources/resources/uicomponents/PhenotypeSelectionUtils.js]: in this entire pull request sometimes a URL like `"/rest/vocabularies/hpo/" + id` is used, and sometimes `"/rest/vocabularies/hpo?input =id`, whats the difference and why\when do we use one and why\when the other?

Comment (allasm) [inline components/widgets/resources/src/main/resources/resources/uicomponents/PhenotypeSelectionUtils.js]: scratch that, it is `hpo/suggest?input=` and `hpo/id`, that makes sense. Just the old code seems to used the same URL schema for both searches and queries of a single term by ID

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml (hunk 1 above)
Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml]: right

Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml]: then I guess I have to rename it everywhere else? That's alot of renaming

File: components/pedigree/resources/src/main/resources/pedigree/hpoTerm.js (hunk 3 above)
Comment (veronikaslc) [inline components/pedigree/resources/src/main/resources/pedigree/hpoTerm.js]: that is addressed in [PR for PT-1403](https://github.com/phenotips/phenotips/pull/2293/commits/4fb308633c7c1c356bd91b18c2c9d3e3634bd547)

**Comment by veronikaslc** [comment]
closing this in favout of  [PR for PT-1403](https://github.com/phenotips/phenotips/pull/2293/commits/4fb308633c7c1c356bd91b18c2c9d3e3634bd547) where I have cherry-picked  commits from this PR.
